### PR TITLE
Issue #1386 Run all oc commands from the VM

### PR DESCRIPTION
**Pull request by praveenkumar**
Currently running oc commands from the host need dns to be working,
and something it doesn't work as expected, running all oc commands
from the VM to make sure that atleast initial startup doesn't depend
on the host side.

### CODE REVIEW

**Comment by praveenkumar** [comment]
Need to update the tests.
/hold

File: pkg/crc/oc/oc.go [hunk 1]
```diff
@@ -82,3 +85,39 @@ func (oc OcConfig) RunOcCommandPrivate(args ...string) (string, string, error) {
...
 	return oc.Runner.RunPrivate(args...)
 }
+
+type OcVMConfig struct {
+	Runner  *ssh.SSHRunner
+	Context string
+	Cluster string
+}
+
+func UseOCWithVMConfig(sshRunner *ssh.SSHRunner) OcVMConfig {
```
Comment (guillaumerose) [inline pkg/crc/oc/oc.go:95]: You can avoid to define this new type by creating a new Runner (and maybe remove use of stderr?).

``` 
type OcSSHRunner struct {
	Runner *ssh.SSHRunner
}

func (r OcSSHRunner) Run(args ...string) (string, string, error) {
	command := fmt.Sprintf("r --kubeconfig /opt/kubeconfig %s", strings.Join(args, " "))
	stdout, err := r.Runner.Run(command)
	return stdout, "", err
}

func (r OcSSHRunner) RunPrivate(args ...string) (string, string, error) {
	command := fmt.Sprintf("r --kubeconfig /opt/kubeconfig %s", strings.Join(args, " "))
	stdout, err := r.Runner.RunPrivate(command)
	return stdout, "", err
}

func UseOCWithSSH(ssh *ssh.SSHRunner) OcConfig {
	return OcConfig{
		Runner: OcSSHRunner{
			Runner: ssh,
		},
		Context: constants.DefaultContext,
		Cluster: constants.DefaultName,
	}
}
```

**Comment by guillaumerose** [comment]
What if `oc` prints something both on stdout and stderr and we need to parse it as json ? Maybe it never happens.

**Pull request merged**